frontend/src/metabase/components/InputBlurChange/InputBlurChange.tsx

```diff
@@ -37,13 +37,22 @@ const InputBlurChange = ({
   const handleBlur = useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
       if (onBlurChange && (value || "") !== event.target.value) {
-        onBlurChange(event);
+        onBlurChange(event.target.value);
       }
     },
     [value, onBlurChange],
   );
 
-  const inputProps = _.omit(props, "onBlur", "onBlurChange", "onChange");
+  useUnmount(() => {
+    const lastPropsValue = value || "";
+    const currentValue = internalValue || "";
+
+    if (onBlurChange && lastPropsValue !== currentValue) {
+      onBlurChange(currentValue);
+    }
+  });
+
+  const inputProps = _.omit(restProps, "onBlur", "onBlurChange", "onChange");
```

Comment:
It seems that `restProps` will never contain such props? Based on this `const { value, onChange, onBlurChange, ...restProps } = props;`